File: app/Models/Review.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Review extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'saved_road_id',
        'rating',
        'comment'
    ];

    public function road()
    {
        return $this->belongsTo(SavedRoad::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
public function photos()
    {
        return $this->hasMany(ReviewPhoto::class);
    }
}

File: android-native/app/src/main/java/com/scenicroutes/app/data/model/Review.kt
package com.scenicroutes.app.data.model

data class Review(
    val id: Long,
    val user_id: Long,
    val saved_road_id: Long,
    val rating: Int,
    val comment: String? = null,
    val created_at: String,
    val updated_at: String,
    val user: User? = null,
    val photos: List<ReviewPhoto>? = null,
)

data class ReviewPhoto(
    val id: Long,
    val review_id: Long,
    val photo_path: String,
    val url: String? = null,
    val thumbnail_url: String? = null,
    val caption: String? = null,
)
